packages/std/src/math/decimal.rs

```diff
@@ -1134,6 +1151,20 @@ mod tests {
         );
     }
 
+    #[test]
+    fn decimal_iter_sum() {
+        let items = vec![
+            Decimal::zero(),
+            Decimal(Uint128::from(2u128)),
+            Decimal(Uint128::from(2u128)),
+        ];
+        assert_eq!(items.iter().sum::<Decimal>(), Decimal(Uint128::from(4u128)));
+        assert_eq!(
+            items.into_iter().sum::<Decimal>(),
+            Decimal(Uint128::from(4u128))
+        )
```

Comment:
Could you add a test with an empty list (can be in the same test function)?

Reply:
Good idea - done.